src/lib_context/context.ml

```diff
@@ -675,6 +675,7 @@ module Make (Encoding : module type of Tezos_context_encoding.Context) = struct
            ~lru_size:!lru_size
            root)
     in
+    Context_metrics.collect () ;
```

Comment:
I'm not fan of having this called from `init`. Could this be placed at topelevel? Or could `Context_metrics.collect` be called from elsewhere in tezose?

Reply:
Did you mean `if not readonly`?

Reply:
Done.